app/javascript/gobierto_data/webapp/components/commons/DownloadButton.vue

```diff
@@ -13,37 +13,58 @@
         name="fade"
         mode="out-in"
       >
-        <div
-          v-show="!isHidden"
-          class="gobierto-data-btn-download-data-modal"
-        >
-          <template
-            v-for="(item, key, index) in arrayFormats"
+        <template v-if="!editor">
+          <div
+            v-show="!isHidden"
+            class="gobierto-data-btn-download-data-modal"
           >
-            <a
-              :key="index"
-              :item="item"
-              :href="editor ? sqlfileCSV : item"
-              :download="titleDataset"
-              class="gobierto-data-btn-download-data-modal-element"
+            <template
+              v-for="(item, key, index) in arrayFormats"
             >
-              {{ key }}
-            </a>
-          </template>
-        </div>
+              <a
+                :key="index"
+                :href="item"
+                :download="titleFile"
+                class="gobierto-data-btn-download-data-modal-element"
+              >
+                {{ key }}
+              </a>
+            </template>
+          </div>
+        </template>
+        <template v-else>
+          <div
+            v-show="!isHidden"
+            class="gobierto-data-btn-download-data-modal"
+          >
+            <template
+              v-for="({ url, name, label }, index) in arrayFormatsQuery"
+            >
+              <a
+                :key="index"
+                class="gobierto-data-btn-download-data-modal-element"
+                @click.prevent="getFiles(url, name)"
+              >
+                {{ label }}
+              </a>
+            </template>
+          </div>
+        </template>
       </transition>
     </Button>
   </div>
 </template>
 <script>
 import Button from "./Button.vue";
 import { baseUrl, CommonsMixin } from "./../../../lib/commons.js";
+import { DownloadFilesFactoryMixin } from "./../../../lib/factories/download";
+
 export default {
   name: 'DownloadButton',
   components: {
     Button
   },
-  mixins: [CommonsMixin],
+  mixins: [CommonsMixin, DownloadFilesFactoryMixin],
```

Comment:
Why did we need the CommonsMixin here?

Reply:
Yes. We need `clickoutside` [to close the modal.](https://github.com/PopulateTools/gobierto/blob/640ce135e5e0c19cbe31b1def6178fa40770d491/app/javascript/gobierto_data/webapp/components/commons/DownloadButton.vue#L4)